src/sentry/integrations/gitlab/integration.py

```diff
@@ -248,23 +246,27 @@ def build_integration(self, state):
         group = self.get_group_info(data['access_token'], state['installation_data'])
         scopes = sorted(GitlabIdentityProvider.oauth_scopes)
         base_url = state['installation_data']['url']
-        domain_name = '%s/%s' % (re.sub(r'https?://', '', base_url), group['path'])
+
+        hostname = urlparse(base_url).netloc
         verify_ssl = state['installation_data']['verify_ssl']
 
         integration = {
             'name': group['name'],
-            'external_id': u'{}:{}'.format(urlparse(base_url).netloc, group['id']),
+            # We splice the host & secret together to create an external id.
+            # This id is used as a webhook secret so we can find the matching
+            # sentry org later on.
+            'external_id': u'{}:{}'.format(hostname, uuid4().hex),
```

Comment:
Should be hostname + group_id instead

Reply:
That would also be used as the 'secret' for webhooks then. GitLab webhooks don't contain any other data that lets us track back to the sentry organization.